app/lib/features/pins/widgets/pin_item.dart

```diff
@@ -107,90 +122,268 @@ class _PinItemState extends ConsumerState<PinItem> {
     );
   }
 
-  // pin content widget
-  Widget _buildPinDescription() {
+// attachment list UI
+  Widget _buildAttachmentList() {
+    final noAttachmentTextStyle = Theme.of(context)
+        .textTheme
+        .labelMedium!
+        .copyWith(color: Theme.of(context).colorScheme.neutral5);
+    final attachmentTitleTextStyle = Theme.of(context).textTheme.labelLarge;
```

Comment:
I thought the plan was to stop doing color-code and style setting in the widgets directly ...?!?

Reply:
That'd make it a bit uglier than it seems. But I'm okay to use default theme colors. Also I think kumar mentioned to only apply custom styling if required for specific components.